source/MaterialXShaderGen/Syntax.cpp

```diff
@@ -149,4 +149,27 @@ namespace MaterialX
 
         return result;
     }
+
+    const string DataType::BOOLEAN = "boolean";
```

Comment:
The majority of these strings are also available in Value.h, in the form of TypedValue<T>::TYPE.  Going forward, we may want to wrap that TypedValue<T> access pattern with something more elegant (while still evaluating to a constant expression), allowing the shared definitions to be leveraged here as well.